DOC add example to docstring of make_moons

likeajumprope opened the pull request:
Adding an example to the make_moons function, in response to #27982


#### What does this implement/fix? Explain your changes.
This PR adds an example to the make_moons function in sklearn.datasets._samples_generator.py. It checks off the make_moons tickmark in #27982

likeajumprope committed:
Add Examples to make_moons

Adding an example to the make_moons functions, in response to #27982

lamdang2k commented on sklearn/datasets/_samples_generator.py:
```diff
@@ -864,6 +864,17 @@ def make_moons(n_samples=100, *, shuffle=True, noise=None, random_state=None):
+    Examples
+    --------
+    >>> from sklearn.datasets import make_moons
+    >>> X, y = make_moons(n_samples = 200, noise =0.2, random_state = 2)
```
`>>> X, y = make_moons(n_samples=200, noise=0.2, random_state=42) `

lamdang2k commented on sklearn/datasets/_samples_generator.py:
```diff
@@ -864,6 +864,17 @@ def make_moons(n_samples=100, *, shuffle=True, noise=None, random_state=None):
+    >>> y.shape
+    (200,)
+    >>> list(y[:10])
+    [1, 0, 1, 1, 1, 0, 0, 0, 1, 0]
```
```[0, 0, 0, 1, 1, 1, 0, 1, 1, 0]```

lamdang2k reviewed (changes requested):
Thanks for the PR. Here are some suggestions regarding syntax/formatting. Regarding other examples, there is no space between parameter and its value. And I think ```random_state=42``` is preferred.

likeajumprope committed:
Remove spaces and set seed to 42

likeajumprope committed:
Update output based on new seed

likeajumprope commented:
Hi @lamdang2k, thank you for your feedback. I have updated the PR according to your suggestions (removed the spaces, updated the seed number and the output accordingly)

likeajumprope committed:
Merge branch 'main' into main

glemaitre committed:
Merge remote-tracking branch 'origin/main' into pr/likeajumprope/28450

glemaitre committed:
show only shape

glemaitre reviewed (approved):
LGTM. Thanks @likeajumprope 

@likeajumprope Next time that you do a pull-request, you should branch from `main` to a development branch instead of modifying your `main` branch. This would avoid some issue later on.

glemaitre commented:
Enabling auto-merge that will merge this PR when the continuous integration will pass.

likeajumprope commented:
> LGTM. Thanks @likeajumprope
> 
> @likeajumprope Next time that you do a pull-request, you should branch from `main` to a development branch instead of modifying your `main` branch. This would avoid some issue later on.

Thanks for the feedback, will do!
